CRM/Contribute/BAO/Contribution.php

```diff
@@ -3508,14 +3508,15 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
           'entity_table' => 'civicrm_financial_item',
         );
         foreach ($params['line_item'] as $fieldId => $fields) {
```

Comment:
Actually, I'd prefer to see $fieldID changed to $lineItemID and $fields to $lineItems too.